RubeesEat/Startup.cs

```diff
@@ -1,20 +1,48 @@
 ﻿using System.Collections.Immutable;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
 using RubeesEat.Model;
 using RubeesEat.Model.DB;
+using IHostingEnvironment = Microsoft.AspNetCore.Hosting.IHostingEnvironment;
 
 namespace RubeesEat;
 
 public class Startup
 {
     public IConfiguration Configuration { get; }
+    private readonly IHostingEnvironment _environment;
 
-    public Startup(IConfiguration configuration)
+    public Startup(IConfiguration configuration, IHostingEnvironment environment)
     {
         Configuration = configuration;
+        _environment = environment;
     }
 
     public void ConfigureServices(IServiceCollection services)
     {
+        if (!_environment.IsEnvironment("Test"))
+        {
+            services.AddAuthentication(options =>
+                    {
+                        options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                        options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+                    })
+                    .AddCookie()
+                    .AddOpenIdConnect(options =>
+                    {
+                        options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+                        options.ClientId = Configuration["OpenId:ClientId"];
+                        options.ClientSecret = Configuration["OpenId:ClientSecret"];
+                        options.Authority = Configuration["OpenId:Authority"];

```

Comment:
Please move all these settings into a class `OpenIDSettings` that we read from the configuration. This way, we can add validation attributes to ensure that all the fields are set correctly. Right now, we hope that the settings are correct.

Reply:
I could put the Required attribute for validation on the fields of the class, if thats what you mean. I don't know what other validation attributes i can set in this case.